File: core/src/main/java/com/cangwang/core/ModuleClient.java
package com.cangwang.core;

import java.util.ArrayList;

/**
 * Created by zjl on 2016/11/23.
 */

public class ModuleClient {
    Class<?> clientClass;
    ArrayList<Object> clientList;

    public ModuleClient(Class<?> clientClass,ArrayList<Object> clients){
        this.clientClass = clientClass;
        this.clientList = clients;
    }
    public ModuleClient(Class<?> clientClass){
        this.clientClass = clientClass;
    }

    public void addClient(Object c){
        if (clientList == null)
            clientList = new ArrayList<>();
        clientList.add(c);
    }
}

File: core/src/main/java/com/cangwang/core/ModuleClient.kt
package com.cangwang.core

import java.util.ArrayList

/**
 * Created by zjl on 2016/11/23.
 */

class ModuleClient {
    internal var clientClass: Class<*>
    internal var clientList: ArrayList<Any>? = null

    constructor(clientClass: Class<*>, clients: ArrayList<Any>) {
        this.clientClass = clientClass
        this.clientList = clients
    }

    constructor(clientClass: Class<*>) {
        this.clientClass = clientClass
    }

    fun addClient(c: Any) {
        if (clientList == null)
            clientList = ArrayList()
        clientList!!.add(c)
    }
}
